packages/theme-check-common/src/checks/deprecated-content-for-arguments/index.ts

```diff
@@ -0,0 +1,71 @@
+import { ContentForMarkup, NodeTypes } from '@shopify/liquid-html-parser';
+import { LiquidCheckDefinition, Severity, SourceCodeType } from '../../types';
+
+// content_for "block" and content_for "blocks" only allow `closest.*` kwargs. We used to accept
+// `context.*` during dev preview. 
+const isDeprecatedContextArgument = (argName: string) => argName.startsWith('context.');
+
+export const DeprecatedContentForArguments: LiquidCheckDefinition = {
+  meta: {
+    code: 'DeprecatedContentForArguments',
+    name: 'Prevent the use of deprecated context argument to the content_for tag',
+    docs: {
+      description:
+        'This check is aimed at preventing the use of deprecated arguments for the content_for tag.',
+      url: 'https://shopify.dev/docs/storefronts/themes/tools/theme-check/checks/deprecated-content-for-arguments',
+      recommended: true,
+    },
+    type: SourceCodeType.LiquidHtml,
+    severity: Severity.WARNING,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    const validationStrategies = {
+      blocks: (node: ContentForMarkup) => {
+        const problematicArguments = node.args.filter((arg) => isDeprecatedContextArgument(arg.name));
+        for (const arg of problematicArguments) {
+          context.report({
+            message: `{% content_for "blocks" %} only accepts 'closest.*' arguments. The 'context.*' arguments usage has been deprecated.`,
+            startIndex: arg.position.start,
+            endIndex: arg.position.end,
+          });
+        }
+      },
+
+      block: (node: ContentForMarkup) => {
+        const problematicArguments = node.args.filter(
+          (arg) => (isDeprecatedContextArgument(arg.name)),
+        );
+
+        for (const arg of problematicArguments) {
+          context.report({
+            message: `{% content_for "block" %} only accepts 'id', 'type' and 'closest.*' arguments.  The 'context.*' arguments usage has been deprecated.`,
+            startIndex: arg.position.start,
+            endIndex: arg.position.end,
+          });
+        }
+      },
+    };
+
+    return {
+
+      async LiquidTag(node) {
+
+        if (node.name !== 'content_for' || typeof node.markup === 'string') {
+          return;
+        }
+        /** "block", "blocks", etc. */
+        const contentForType = node.markup.contentForType.value;
+        const validate = validationStrategies[contentForType as keyof typeof validationStrategies];
+        if (!validate) {
+          return;
+        }
+
+      
+        validate(node.markup);
+      },
+    };
+  },
+};
```

Comment:
Given how transient this thing is and the thing hasn't been released. We can probably not ship this. Just let the one that accepts `closest` complain about `context` not being supported.

Reply:
Thing is this could live in this repo forever. People will start being able to have that in their `.theme-check.yml` and I can't remember what the behaviour is for when you have settings for checks that don't exist. 

Which would probably mean that removing the check would require a major version bump of theme-check. 

Do we have another check that we could use instead? Something that wouldn't go away? Or maybe something like DeprecatedFeature?

Reply:
That's probably a nice place for it yeah!

Reply:
dropped this check